### PR TITLE
brochure and lan on homepage

### PR DESCRIPTION
**Pull request by AlokSP**

### CODE REVIEW

File: src/main/java/com/health/threadpool/TaskProcessingService.java [hunk 1]
```diff
@@ -326,7 +326,8 @@ public void addUpdateDeleteTutorial(Tutorial tutorial, String requestType) {
             outlinePath = Optional.of(tutorial.getOutlinePath());
         }
 
-        if (requestType.equals(CommonData.ADD_DOCUMENT) && !tutorial.isAddedQueue() && !sm_url.isEmpty()) {
+        if (requestType.equals(CommonData.ADD_DOCUMENT) && !tutorial.isAddedQueue() && !sm_url.isEmpty()
+                && isURLWorking(documentPathforOriginalScript)) {
```
Comment (sunilshetye) [inline src/main/java/com/health/threadpool/TaskProcessingService.java:330]: don't call isURLWorking twice for the same URL. assign to a boolean variable once and use it.

File: src/main/java/com/health/threadpool/TaskProcessingService.java [hunk 2]
```diff
@@ -355,7 +356,7 @@ public void addUpdateDeleteTutorial(Tutorial tutorial, String requestType) {
 
         }
 
-        else if (tutorial.isAddedQueue()) {
+        else if (tutorial.isAddedQueue() && isURLWorking(documentPathforOriginalScript)) {
```
Comment (sunilshetye) [inline src/main/java/com/health/threadpool/TaskProcessingService.java:359]: don't call isURLWorking twice for the same URL. assign to a boolean variable once and use it.

File: src/main/java/com/health/controller/HomeController.java [hunk 3]
```diff
@@ -7230,13 +7232,17 @@ public String brochure(HttpServletRequest req, Principal principal, Model model)
...
                 FilesofBrouchure filesBro = filesofbrouchureService.findByLanguageandVersion(lan, ver1);
                 if (filesBro != null) {
+                    brochurelanguages.add(lan);
                     filesList.add(filesBro);
                 }
             }
         }
 
+        languages = new ArrayList<>(brochurelanguages);
+        languages.sort(Comparator.comparing(Language::getLangName));
```
Comment (sunilshetye) [inline src/main/java/com/health/controller/HomeController.java:7245]: What is this sorting for?

**Review by sunilshetye — CHANGES_REQUESTED** [review]

File: src/main/java/com/health/controller/HomeController.java (hunk 3 above)
Comment (AlokSP) [reply src/main/java/com/health/controller/HomeController.java:7245]: it sorts language by name for the user.

Comment (sunilshetye) [reply src/main/java/com/health/controller/HomeController.java:7245]: okay, thanks

**Review by sunilshetye — APPROVED** [review]

**Pull request merged**